### PR TITLE
Support for git submodules.

### PR DESCRIPTION
**Pull request by badc0re**
## What changes are proposed in this pull request?

Added support for pulling git sub-modules for project.

## How is this patch tested?

Tested on a git repository with a sub-module on a specific branch.
Also tested with the small test set (./travis/run-small-python-tests.sh).

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

When running a git project, if the project has a sub-module/s on master or any branch,
they will be pulled.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [x] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by Borda — APPROVED** [review]
looks fine... :)

**Comment by badc0re** [comment]
> looks fine... :)

Can you merge? :)

File: mlflow/projects/__init__.py [hunk 1]
```diff
@@ -390,6 +390,7 @@ def _fetch_git_repo(uri, version, dst_dir):
     else:
         repo.create_head("master", origin.refs.master)
         repo.heads.master.checkout()
+    repo.git.submodule("update", "--init")
```
Comment (dbczumar) [inline mlflow/projects/__init__.py]: @badc0re Can we make this submodule initialization recursive?

**Review by dbczumar — COMMENTED** [review]
@badc0re Thanks for the contribution! One minor question about recursive submodule initialization.

**Review by dbczumar — APPROVED** [review]
@badc0re I went ahead and swapped in a call to the `repo.submodule_update`, explicitly specifying recursive initialization. This LGTM! Will merge when tests pass. Thanks for calling this out!

**Comment by badc0re** [comment]
> @badc0re I went ahead and swapped in a call to the `repo.submodule_update`, explicitly specifying recursive initialization. This LGTM! Will merge when tests pass. Thanks for calling this out!

Great thanks, I was waiting for the weekend to change it, but you were faster! :D

**Pull request merged**